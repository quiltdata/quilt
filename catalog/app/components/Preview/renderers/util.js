import cx from 'classnames'
import * as React from 'react'
import * as M from '@material-ui/core'

const useMsgStyles = M.makeStyles((t) => ({
  root: {
    borderRadius: t.shape.borderRadius,
    padding: t.spacing(1.5),
    whiteSpace: 'pre-wrap',
    ...t.typography.body2,
  },
  inner: {
    maxHeight: 100,
    overflow: 'auto',
  },
  info: {
    background: t.palette.info.light,
    color: t.palette.info.contrastText,
  },
  warning: {
    background: t.palette.warning.light,
    color: t.palette.warning.contrastText,
  },
}))

export function Msg({ type = 'info', className, children, ...props }) {
  const classes = useMsgStyles()
  return (
    <M.Box className={cx(classes.root, classes[type], className)} {...props}>
      <div className={classes.inner}>{children}</div>
    </M.Box>
  )
}

export const renderPreviewStatus = ({ note, warnings }) => (
  <Msg type="warning" mb={2}>
    {!!note && note}
    {!!note && !!warnings && '\n\n'}
    {!!warnings && warnings}
  </Msg>
)
