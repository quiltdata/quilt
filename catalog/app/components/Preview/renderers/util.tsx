import cx from 'classnames'
import * as React from 'react'
import * as M from '@material-ui/core'

const useMsgStyles = M.makeStyles((t) => ({
  root: {
    borderRadius: t.shape.borderRadius,
    padding: t.spacing(1.5),
    whiteSpace: 'pre-wrap',
    ...t.typography.body2,
  },
  inner: {
    maxHeight: 100,
    overflow: 'auto',
  },
  info: {
    background: t.palette.info.main,
    color: t.palette.info.contrastText,
  },
  warning: {
    background: t.palette.warning.light,
    color: t.palette.warning.contrastText,
  },
}))

interface MsgProps extends M.BoxProps {
  type: 'info' | 'warning'
  className: string
}

export function Msg({ type = 'info', className, children, ...props }: MsgProps) {
  const classes = useMsgStyles()
  return (
    <M.Box className={cx(classes.root, classes[type], className)} {...props}>
      <div className={classes.inner}>{children}</div>
    </M.Box>
  )
}

const useMsgAccordionStyles = M.makeStyles((t) => ({
  root: {
    borderRadius: t.shape.borderRadius,
    marginBottom: t.spacing(2),
  },
  icon: {
    marginRight: '12px',
  },
  msg: {
    width: '100%',
  },
  header: {
    width: '100%',
    alignItems: 'center',
    display: 'flex',
  },
}))

interface MsgAccordionProps {
  title: string
  type: 'info' | 'warning'
  warnings: string
}

function MsgAccordion({ title, type, warnings }: MsgAccordionProps) {
  const classes = useMsgAccordionStyles()
  const warningsCount = warnings.match(/\n/g)?.length
  return (
    <M.Accordion className={classes.root} variant="outlined">
      <M.AccordionSummary expandIcon={<M.Icon>expand_more</M.Icon>}>
        <div className={classes.header}>
          {warningsCount && <M.Chip className={classes.icon} label={warningsCount} />}
          {title}
        </div>
      </M.AccordionSummary>
      <M.AccordionDetails>
        <Msg className={classes.msg} type={type}>
          {warnings}
        </Msg>
      </M.AccordionDetails>
    </M.Accordion>
  )
}

export const renderWarnings = (warnings?: string) => {
  if (!warnings) return null

  return <MsgAccordion title="Parsing errors" type="warning" warnings={warnings} />
}
