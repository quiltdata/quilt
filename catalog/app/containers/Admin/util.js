import mapValues from 'lodash/mapValues';
import omit from 'lodash/omit';
import { compose, withHandlers, withProps, withStateHandlers } from 'recompose';

export const formatDate = (d) => d ? new Date(d).toLocaleString() : 'N/A';

export const withStatefulActions = (bindActions) => compose(
  withStateHandlers({ pending: {} }, {
    startAction: ({ pending }) => (key) => ({
      pending: Object.assign({}, pending, { [key]: true }),
    }),
    endAction: ({ pending }) => (key) => ({
      pending: omit(pending, [key]),
    }),
  }),
  withHandlers(({ actions }) =>
    mapValues(actions, (action, key) =>
      // eslint-disable-next-line object-curly-newline
      ({ pending, startAction, endAction, actions: { [key]: callback } }) => (arg) => {
        if (pending[arg]) return;
        startAction(arg);
        callback(arg)
          .then(() => endAction(arg))
          .catch((e) => { endAction(arg); throw e; });
      })),
  withProps((props) => ({ bindActions: bindActions(props) })),
);
