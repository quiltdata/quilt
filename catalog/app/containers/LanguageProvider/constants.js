/*
 *
 * LanguageProvider constants
 *
 */

export const REDUX_KEY = 'app/LanguageProvider';

export const CHANGE_LOCALE = 'app/LanguageProvider/CHANGE_LOCALE';
export const DEFAULT_LOCALE = 'en';
