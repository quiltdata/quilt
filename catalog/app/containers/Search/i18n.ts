// Descriptive filter labels
export const PACKAGE_FILTER_LABELS = {
  bucket: 'Bucket',
  comment: 'Package revision comment',
  entries: 'Total number of files in the package',
  hash: 'Package hash',
  modified: 'Last modified',
  name: 'Package name',
  size: 'Cumulative package size',
  workflow: 'Workflow',
}

// Shortened filter labels
export const COLUMN_LABELS = {
  bucket: 'Bucket',
  comment: 'Comment',
  entries: 'File count',
  hash: 'Hash',
  modified: 'Modified',
  name: 'Package',
  size: 'Size',
  workflow: 'Workflow',
}

export const OBJECT_FILTER_LABELS = {
  content: 'Contents',
  deleted: 'Delete marker',
  ext: 'Extension',
  key: 'Object key',
  modified: 'Last modified',
  size: 'Object size',
}
