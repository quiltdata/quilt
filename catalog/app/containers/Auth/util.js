import { stringify } from 'query-string';

import { LATENCY_SECONDS } from './constants';

export const makeHeadersFromTokens = ({ token }) => ({
  Authorization: `Bearer ${token}`,
});

const makeURL = (url, query) => {
  const qs = stringify(query);
  const search = qs ? `?${qs}` : '';
  return `${url}${search}`;
};

const getPath = (loc) => loc.pathname + loc.search;

export const makeSignInURL = (next = getPath(window.location)) =>
  // TODO(dima): Sign up vs sign in?
  makeURL('/signin', { next });

export const adjustTokensForLatency = (tokens) => ({
  ...tokens,
  expires_at:
    Number.isFinite(tokens.expires_at)
      ? tokens.expires_at - LATENCY_SECONDS
      : tokens.expires_at,
});
