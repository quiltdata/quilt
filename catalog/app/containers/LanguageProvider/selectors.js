import { createSelector } from 'reselect';

import { REDUX_KEY } from './constants';

/**
 * Direct selector to the languageToggle state domain
 */
const selectLanguage = (state) => state.get(REDUX_KEY);

/**
 * Select the language locale
 */

const makeSelectLocale = () => createSelector(
  selectLanguage,
  (languageState) => languageState.get('locale')
);

export {
  selectLanguage,
  makeSelectLocale,
};
