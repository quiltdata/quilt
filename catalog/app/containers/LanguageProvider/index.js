/*
 *
 * LanguageProvider
 *
 * this component connects the redux state language locale to the
 * IntlProvider component and i18n messages (loaded from `app/translations`)
 */

import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { mapProps, setPropTypes } from 'recompose';
import { createSelector } from 'reselect';
import { IntlProvider } from 'react-intl';

import { composeComponent } from 'utils/reactTools';
import { injectReducer } from 'utils/ReducerInjector';

import { REDUX_KEY } from './constants';
import reducer from './reducer';
import { makeSelectLocale } from './selectors';

export default composeComponent('LanguageProvider',
  injectReducer(REDUX_KEY, reducer),
  connect(createSelector(
    makeSelectLocale(),
    (locale) => ({ locale })
  )),
  setPropTypes({
    locale: PropTypes.string,
    messages: PropTypes.object,
    children: PropTypes.element.isRequired,
  }),
  mapProps(({ locale, messages, ...props }) => ({
    locale,
    key: locale,
    messages: messages[locale],
    ...props,
  })),
  IntlProvider);
