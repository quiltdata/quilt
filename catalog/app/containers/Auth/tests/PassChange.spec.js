import createHistory from 'history/createMemoryHistory';
import { fromJS } from 'immutable';
import React from 'react';

import LanguageProvider from 'containers/LanguageProvider';
import { translationMessages as messages } from 'i18n';
import FormProvider from 'utils/ReduxFormProvider';
import StoreProvider from 'utils/StoreProvider';
import configureStore from 'store';

import feature from 'testing/feature';
import { mockComponentSelector } from 'testing/util';

import {
  Provider as AuthProvider,
  PassChange,
  errors,
} from '..';

import { api } from './support/fixtures';
import errorSteps from './support/error';
import formSteps from './support/form';
import reactSteps from './support/react';
import requestsSteps from './support/requests';

jest.mock('material-ui/RaisedButton');
jest.mock('material-ui/TextField');
jest.mock('react-router-dom');

jest.mock('components/Spinner');
jest.mock('constants/config');
jest.mock('utils/errorReporting');
jest.mock('utils/time');

const requests = {
  changePassword: {
    setup: () => ['post', '/reset_password'],
    expect: ({ link, password }) =>
      expect.objectContaining({
        body: JSON.stringify({ link, password }),
      }),
  },
};

const values = {
  password: 'valid',
  'invalid password': 'invalid',
};

const fields = ['password', 'passwordCheck'];

const onSubmit = (ctx) => ({
  ...ctx,
  password:
    ctx.mounted
      .find(`${mockComponentSelector('TextField')}[name="password"]`)
      .prop('value'),
});

const screens = {
  success: (html) => {
    const msgs = html.find('p');
    expect(msgs.eq(0).text()).toMatch('Your password has been changed');
    expect(msgs.eq(1).text()).toMatch('Now you can sign in');
    expect(msgs.eq(1).find(mockComponentSelector('Link')).attr('to')).toBe('/signin');
  },
};

const setup = () => {
  const history = createHistory({ initialEntries: ['/'] });
  const store = configureStore(fromJS({}), history);
  const storage = {
    load: () => ({}),
    set: () => {},
    remove: () => {},
  };
  const link = 'test-link';
  const tree = (
    <StoreProvider store={store}>
      <FormProvider>
        <LanguageProvider messages={messages}>
          <AuthProvider
            storage={storage}
            api={api}
          >
            <PassChange match={{ params: { link } }} />
          </AuthProvider>
        </LanguageProvider>
      </FormProvider>
    </StoreProvider>
  );
  return { tree, link };
};

const steps = [
  ...requestsSteps(requests),
  ...formSteps({ values, fields, onSubmit }),
  ...errorSteps(errors.AuthError),
  ...reactSteps({ setup, screens }),
];


feature('containers/Auth/PassChange')
  .scenario('Trying to change the password when the link is invalid')

  .given('changePassword request is expected')

  .when('the component tree is mounted')
  .then('I should see the form in interactive state')
  .then('the rendered markup should match the snapshot')

  .when('I submit the form')
  .then('I should see the form in invalid state')
  .then('the rendered markup should match the snapshot')
  .then('I should see error on password field: "Enter a password"')
  .then('I should see error on passwordCheck field: "Enter the password again"')

  .when('I enter password into password field')
  .then('I should see no error on password field')

  .when('I enter invalid password into passwordCheck field')
  .then('I should see error on passwordCheck field: "Passwords must match"')

  .when('I enter password into passwordCheck field')
  .then('I should see no error on passwordCheck field')
  .then('I should see the form in valid state')

  .when('I submit the form')
  .then('I should see the form in waiting state')
  .then('changePassword request should be made')

  .when('changePassword request fails with 401, error: "Reset token invalid."')
  .then('I should see the form in invalid state')
  .then('the rendered markup should match the snapshot')
  .then('I should see form error: "This reset link is invalid"')

  .back()
  .when('changePassword request fails with 404, error: "User not found."')
  .then('I should see the form in invalid state')
  .then('the rendered markup should match the snapshot')
  .then('I should see form error: "This reset link is invalid"')

  .back()
  .when('changePassword request fails with 500')
  .then('I should see the form in invalid state')
  .then('the rendered markup should match the snapshot')
  .then('I should see form error: "Something went wrong"')
  .then('the error should be captured')


  .scenario('Changing the password when the link is valid')

  .given('changePassword request is expected')

  .when('the component tree is mounted')
  .when('I enter invalid password into password field')
  .when('I enter invalid password into passwordCheck field')
  .when('I submit the form')
  .then('I should see the form in waiting state')
  .then('changePassword request should be made')

  .when('changePassword request fails with 400, message: "Password must be"')
  .then('I should see the form in invalid state')
  .then('the rendered markup should match the snapshot')
  .then('I should see error on password field: "Password must be"')

  .when('I enter password into password field')
  .then('I should see no error on password field')

  .when('I enter password into passwordCheck field')
  .then('I should see no error on passwordCheck field')
  .then('I should see the form in valid state')

  .when('I submit the form')
  .when('changePassword request succeeds')
  .then('I should see the success screen')
  .then('the rendered markup should match the snapshot')

  .steps(steps)
  .run();
