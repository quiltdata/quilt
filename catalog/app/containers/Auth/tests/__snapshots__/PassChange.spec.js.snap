// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: containers/Auth/PassChange Background: Scenario: Changing the password when the link is valid Given changePassword request is expected When the component tree is mounted When I enter invalid password into password field When I enter invalid password into passwordCheck field When I submit the form When changePassword request fails with 400, message: "Password must be" Then the rendered markup should match the snapshot 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  max-width: 280px;
  min-height: calc(100vh - 300px);
  width: 100%;
}

.c1 {
  text-align: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 32px;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    <span>
      Change Password
    </span>
  </h1>
  <form>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="password"
      type="password"
      value="invalid"
    >
      <prop-floating-label-text>
        <span>
          New password
        </span>
      </prop-floating-label-text>
      <prop-error-text>
        <span>
          Password must be at least 8 characters long
        </span>
      </prop-error-text>
    </component-text-field>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="passwordCheck"
      type="password"
      value="invalid"
    >
      <prop-floating-label-text>
        <span>
          Re-enter your new password
        </span>
      </prop-floating-label-text>
    </component-text-field>
    <div
      class="c2"
    >
      <component-raised-button
        disabled="true"
        primary="true"
        type="submit"
      >
        <prop-label>
          <span>
            Change Password
          </span>
        </prop-label>
      </component-raised-button>
    </div>
  </form>
</div>
`;

exports[`Feature: containers/Auth/PassChange Background: Scenario: Changing the password when the link is valid Given changePassword request is expected When the component tree is mounted When I enter invalid password into password field When I enter invalid password into passwordCheck field When I submit the form When changePassword request fails with 400, message: "Password must be" When I enter password into password field When I enter password into passwordCheck field When I submit the form When changePassword request succeeds Then the rendered markup should match the snapshot 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  max-width: 280px;
  min-height: calc(100vh - 300px);
  width: 100%;
}

.c1 {
  text-align: center;
}

.c2 {
  text-align: center;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    <span>
      Change Password
    </span>
  </h1>
  <p
    class="c2"
  >
    <span>
      Your password has been changed.
    </span>
  </p>
  <p
    class="c2"
  >
    <span>
      Now you can 
      <component-link
        to="/signin"
      >
        <span>
          sign in
        </span>
      </component-link>
       using your new password.
    </span>
  </p>
</div>
`;

exports[`Feature: containers/Auth/PassChange Background: Scenario: Trying to change the password when the link is invalid Given changePassword request is expected When the component tree is mounted Then the rendered markup should match the snapshot 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  max-width: 280px;
  min-height: calc(100vh - 300px);
  width: 100%;
}

.c1 {
  text-align: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 32px;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    <span>
      Change Password
    </span>
  </h1>
  <form>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="password"
      type="password"
      value=""
    >
      <prop-floating-label-text>
        <span>
          New password
        </span>
      </prop-floating-label-text>
    </component-text-field>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="passwordCheck"
      type="password"
      value=""
    >
      <prop-floating-label-text>
        <span>
          Re-enter your new password
        </span>
      </prop-floating-label-text>
    </component-text-field>
    <div
      class="c2"
    >
      <component-raised-button
        disabled="false"
        primary="true"
        type="submit"
      >
        <prop-label>
          <span>
            Change Password
          </span>
        </prop-label>
      </component-raised-button>
    </div>
  </form>
</div>
`;

exports[`Feature: containers/Auth/PassChange Background: Scenario: Trying to change the password when the link is invalid Given changePassword request is expected When the component tree is mounted When I submit the form Then the rendered markup should match the snapshot 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  max-width: 280px;
  min-height: calc(100vh - 300px);
  width: 100%;
}

.c1 {
  text-align: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 32px;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    <span>
      Change Password
    </span>
  </h1>
  <form>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="password"
      type="password"
      value=""
    >
      <prop-floating-label-text>
        <span>
          New password
        </span>
      </prop-floating-label-text>
      <prop-error-text>
        <span>
          Enter a password
        </span>
      </prop-error-text>
    </component-text-field>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="passwordCheck"
      type="password"
      value=""
    >
      <prop-floating-label-text>
        <span>
          Re-enter your new password
        </span>
      </prop-floating-label-text>
      <prop-error-text>
        <span>
          Enter the password again
        </span>
      </prop-error-text>
    </component-text-field>
    <div
      class="c2"
    >
      <component-raised-button
        disabled="true"
        primary="true"
        type="submit"
      >
        <prop-label>
          <span>
            Change Password
          </span>
        </prop-label>
      </component-raised-button>
    </div>
  </form>
</div>
`;

exports[`Feature: containers/Auth/PassChange Background: Scenario: Trying to change the password when the link is invalid Given changePassword request is expected When the component tree is mounted When I submit the form When I enter password into password field When I enter invalid password into passwordCheck field When I enter password into passwordCheck field When I submit the form When changePassword request fails with 401, error: "Reset token invalid." Then the rendered markup should match the snapshot 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  max-width: 280px;
  min-height: calc(100vh - 300px);
  width: 100%;
}

.c1 {
  text-align: center;
}

.c2 {
  color: #f44336;
  margin-top: 24px;
  text-align: center;
}

.c2 a {
  color: inherit !important;
  text-decoration: underline;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 32px;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    <span>
      Change Password
    </span>
  </h1>
  <form>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="password"
      type="password"
      value="valid"
    >
      <prop-floating-label-text>
        <span>
          New password
        </span>
      </prop-floating-label-text>
    </component-text-field>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="passwordCheck"
      type="password"
      value="valid"
    >
      <prop-floating-label-text>
        <span>
          Re-enter your new password
        </span>
      </prop-floating-label-text>
    </component-text-field>
    <p
      class="c2"
    >
      <span>
        This reset link is invalid. Try 
        <component-link
          to="/reset_password"
        >
          <span>
            resetting password
          </span>
        </component-link>
         again.
      </span>
    </p>
    <div
      class="c3"
    >
      <component-raised-button
        disabled="true"
        primary="true"
        type="submit"
      >
        <prop-label>
          <span>
            Change Password
          </span>
        </prop-label>
      </component-raised-button>
    </div>
  </form>
</div>
`;

exports[`Feature: containers/Auth/PassChange Background: Scenario: Trying to change the password when the link is invalid Given changePassword request is expected When the component tree is mounted When I submit the form When I enter password into password field When I enter invalid password into passwordCheck field When I enter password into passwordCheck field When I submit the form When changePassword request fails with 404, error: "User not found." Then the rendered markup should match the snapshot 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  max-width: 280px;
  min-height: calc(100vh - 300px);
  width: 100%;
}

.c1 {
  text-align: center;
}

.c2 {
  color: #f44336;
  margin-top: 24px;
  text-align: center;
}

.c2 a {
  color: inherit !important;
  text-decoration: underline;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 32px;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    <span>
      Change Password
    </span>
  </h1>
  <form>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="password"
      type="password"
      value="valid"
    >
      <prop-floating-label-text>
        <span>
          New password
        </span>
      </prop-floating-label-text>
    </component-text-field>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="passwordCheck"
      type="password"
      value="valid"
    >
      <prop-floating-label-text>
        <span>
          Re-enter your new password
        </span>
      </prop-floating-label-text>
    </component-text-field>
    <p
      class="c2"
    >
      <span>
        This reset link is invalid. Try 
        <component-link
          to="/reset_password"
        >
          <span>
            resetting password
          </span>
        </component-link>
         again.
      </span>
    </p>
    <div
      class="c3"
    >
      <component-raised-button
        disabled="true"
        primary="true"
        type="submit"
      >
        <prop-label>
          <span>
            Change Password
          </span>
        </prop-label>
      </component-raised-button>
    </div>
  </form>
</div>
`;

exports[`Feature: containers/Auth/PassChange Background: Scenario: Trying to change the password when the link is invalid Given changePassword request is expected When the component tree is mounted When I submit the form When I enter password into password field When I enter invalid password into passwordCheck field When I enter password into passwordCheck field When I submit the form When changePassword request fails with 500 Then the rendered markup should match the snapshot 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  max-width: 280px;
  min-height: calc(100vh - 300px);
  width: 100%;
}

.c1 {
  text-align: center;
}

.c2 {
  color: #f44336;
  margin-top: 24px;
  text-align: center;
}

.c2 a {
  color: inherit !important;
  text-decoration: underline;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 32px;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    <span>
      Change Password
    </span>
  </h1>
  <form>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="password"
      type="password"
      value="valid"
    >
      <prop-floating-label-text>
        <span>
          New password
        </span>
      </prop-floating-label-text>
    </component-text-field>
    <component-text-field
      disabled="false"
      fullwidth="true"
      name="passwordCheck"
      type="password"
      value="valid"
    >
      <prop-floating-label-text>
        <span>
          Re-enter your new password
        </span>
      </prop-floating-label-text>
    </component-text-field>
    <p
      class="c2"
    >
      <span>
        Something went wrong. Try again later.
      </span>
    </p>
    <div
      class="c3"
    >
      <component-raised-button
        disabled="true"
        primary="true"
        type="submit"
      >
        <prop-label>
          <span>
            Change Password
          </span>
        </prop-label>
      </component-raised-button>
    </div>
  </form>
</div>
`;
